CUSTOM.md

```diff
@@ -0,0 +1,218 @@
+        - blightCard: boolean - whether blight card is being used
+        - expansions: table - expansions included in setup, keys are the name of them
+        - thematic: boolean - whether thematic island setup is enabled
+        - adversary: boolean - whether an adversary is being used (only exists for Scenario call)
+      - return boolean - whether this adversary/scenario should be used
+  - preSetup - called before all other setup steps have started
+    - preSetup: boolean - set to true if you have PreSetup function
+    - preSetupComplete: boolean - set to true at end PreSetup function
+    - PreSetup(params)
+      - params: table - contains data about game setup (only exists for Adversary call)
+        - level: number - level of current adversary
+  - invaderDeckSetup - allows you to modify invader deck
+    - invaderDeckSetup: boolean - set to true if you have InvaderDeckSetup function
+    - invaderDeckSetupComplete: boolean - set to true at end InvaderDeckSetup function
+    - InvaderDeckSetup(params)
+      - params: table - contains data about game setup
+        - level: number - level of current adversary (only exists for Adversary call)
+        - deck: table - table containing current setup for invader deck
+          - values of 1, 2, 3 indicate random card of that stage
+          - values of W, M, J, S indicate specific stage I cards
+          - values of We, Me, Je, Se, C indicate specific stage II cards
+          - values of MW, JW, SW, MJ, SM, JS indicate specific stage III cards
+          - values of <anything above>* will have UI treat card as one stage lower (i.e. Prussia 2)
+        - return table - updated table for invader deck setup
+  - mapSetup - allows you to modify the pieces on a board
+    - mapSetup: boolean - set to true if you have MapSetup function
+    - mapSetupComplete: boolean - set to true at end MapSetup function
+    - MapSetup(params)
+      - params: table - contains data about game setup
+        - level: number - level of current adversary (only exists for Adversary call)
+        - pieces: 2D table - contain keys for each land number, and each land table has values for piece name
+        - guid: string - guid of the board being setup (only exists for Adversary call)
+        - extra: boolean - whether the current board is the extra board from setup variant
+      - return table - updated table for pieces setup
+  - postSetup - called after all other setup steps have completed
+    - postSetup: boolean - set to true if you have PostSetup function
+    - postSetupComplete: boolean - set to true at end PostSetup function
+    - PostSetup(params)
+      - params: table - contains data about game setup (only exists for Adversary call)
+        - level: number - level of current adversary
+        - other: table - contains data about other adversary
+          - exist: boolean - whether there is a second adversary
+          - level: number - other adversary level
+- Adversary Specific:
```

Comment:
This whole section should mimic the order of the adversary (even if it's an artificial order you used on your programing of the regular adversaries) ...